src/libs/actions/PaymentMethods.js

```diff
@@ -109,12 +109,26 @@ function setWalletLinkedAccount(password, bankAccountID, fundID) {
                     walletLinkedAccountID: bankAccountID || fundID, walletLinkedAccountType: bankAccountID ? CONST.PAYMENT_METHODS.BANK_ACCOUNT : CONST.PAYMENT_METHODS.DEBIT_CARD,
                 });
                 Growl.show(Localize.translateLocal('paymentsPage.setDefaultSuccess'), CONST.GROWL.SUCCESS, 5000);
-            } else {
-                Growl.show(Localize.translateLocal('paymentsPage.setDefaultFailure'), CONST.GROWL.ERROR, 5000);
+                return;
+            }
+            Growl.show(Localize.translateLocal('paymentsPage.error.setDefaultFailure'), CONST.GROWL.ERROR, 5000);
+        }).catch((error) => {
+            // Make sure to show user more specific errors which will help support identify the problem faster.
+            switch (error.message) {
+                case CONST.WALLET.ERROR.INVALID_WALLET:
+                case CONST.WALLET.ERROR.NOT_OWNER_OF_BANK_ACCOUNT:
+                    Growl.show(`${Localize.translateLocal('paymentsPage.error.notOwnerOfBankAccount')} ${Localize.translateLocal('common.conciergeHelp')}`, CONST.GROWL.ERROR, 5000);
+                    return;
+                case CONST.WALLET.ERROR.NOT_OWNER_OF_FUND:
+                case CONST.WALLET.ERROR.INVALID_FUND:
+                    Growl.show(`${Localize.translateLocal('paymentsPage.error.notOwnerOfFund')} ${Localize.translateLocal('common.conciergeHelp')}`, CONST.GROWL.ERROR, 5000);
+                    return;
+                case CONST.WALLET.ERROR.INVALID_BANK_ACCOUNT:
+                    Growl.show(`${Localize.translateLocal('paymentsPage.error.invalidBankAccount')} ${Localize.translateLocal('common.conciergeHelp')}`, CONST.GROWL.ERROR, 5000);
+                    return;
+                default:
+                    Growl.show(Localize.translateLocal('paymentsPage.error.setDefaultFailure'), CONST.GROWL.ERROR, 5000);
```

Comment:
Anyone know if it's possible for this API command promise method to land in this `.catch()` block?

Reply:
What exception is thrown and where?

Reply:
Oh I think it's this weird code here that I am removing 😄 

https://github.com/Expensify/App/blob/c26e41573c6b6ea91abef85afc173ba6ecc4cd2c/src/libs/API.js#L182-L187

I think most errors do not work this way and `catch()` should never do anything. No worries though just in the process of cleaning some stuff up.

Reply:
There is this [one](https://github.com/Expensify/Auth/blob/3354b5b8ae4ec12400c871f744395fa51cf180fc/auth/command/SetWalletLinkedAccount.cpp#L35) which we decided not to catch specifically as it should not happen but if it happens, the error message would need to be very general as the default already was.

Reply:
Yes thanks I would like to know where the exception gets thrown in the JavaScript layer.

This issue has more details: https://github.com/Expensify/Expensify/issues/201109

Reply:
Oh interesting, I have first tried to handle it in the `then` case but it did not work for me, `catch` did. Not sure now why that is happening. Catch worked locally.

Reply:
Ah, thanks for the context! 🙌

Reply:
Yeah it's because the promise is getting rejected which will trigger the `.catch()` block and would be similar to throwing an exception inside the `.then()` block. Kind of weird IMO because who knows how many API can throw these codes (or which may throw them in the future) and any that are not handled like you did here will crash the app.

Reply:
Oh, that is weird indeed. We should standardize this for sure.